src/python/pants/backend/python/goals/export.py

```diff
@@ -134,6 +134,44 @@ class ExportPluginOptions:
             """
         ),
         advanced=True,
+        removal_version="2.24.0.dev0",
+        removal_hint=softwrap(
+            """
+            Use `--export-py-hermetic-in-resolve` instead.
+            """
+        )
+    )
+
+    py_hermetic_scripts_in_resolve = StrListOption(
```

Comment:
```suggestion
    py_nonhermetic_scripts_in_resolve = StrListOption(
```

We need to default to hermetic scripts. So maybe this should be the inverse.

Reply:
Yes, I asked about that in Slack: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1720530340619879?thread_ts=1719351964.511499&cid=C0D7TNJHL

But this change wasn't supposed to be in this PR. Closing this PR since I screwed up my git history on this branch. <sigh>